tests/test_financial.py

```diff
@@ -90,13 +93,145 @@ def test_decimal_with_when(self):
+                "pmt": 1000,
+                "fv": 0,
+            },
+            "expected_result": numpy.array(
+                [-9471.30, -9222.18, -8982.59, -8752.06, -8530.20, -8316.61]
+            ),
+        },
+    }
+
+    # Randomized input strategies for fuzz tests & property-based tests
+    numeric_strategy = st.one_of(
+        st.decimals(),
+        st.floats(),
+        st.integers(),
+    )
+
+    when_period_strategy = st.sampled_from(["end", "begin", 1, 0])
+
     def test_pv(self):
         assert_almost_equal(npf.pv(0.07, 20, 12000, 0), -127128.17, 2)
 
     def test_pv_decimal(self):
-        assert_equal(npf.pv(Decimal('0.07'), Decimal('20'), Decimal('12000'),
-                            Decimal('0')),
-                     Decimal('-127128.1709461939327295222005'))
+        assert_equal(
+            npf.pv(Decimal("0.07"), Decimal("20"), Decimal("12000"), Decimal("0")),
+            Decimal("-127128.1709461939327295222005"),
+        )
+
+    @pytest.mark.parametrize("test_case", test_cases.values(), ids=test_cases.keys())
+    def test_pv_examples(self, test_case):
+        inputs, expected_result = test_case["inputs"], test_case["expected_result"]
+        result = npf.pv(**inputs)
+        assert result == pytest.approx(expected_result)
+
+    @pytest.mark.slow
+    @given(
+        rate=numeric_strategy,
+        nper=numeric_strategy,
+        pmt=numeric_strategy,
+        fv=numeric_strategy,
+        when=when_period_strategy,
+    )
+    @settings(verbosity=Verbosity.verbose)
+    def test_pv_fuzz(
+        self,
+        rate: Union[int, float, Decimal, numpy.ndarray],
+        nper: Union[int, float, Decimal, numpy.ndarray],
+        pmt: Union[int, float, Decimal, numpy.ndarray],
+        fv: Union[int, float, Decimal, numpy.ndarray],
+        when: Literal[0, 1, "begin", "end"],
+    ) -> None:
+        npf.pv(rate, nper, pmt, fv, when)
+
+    @pytest.mark.slow
+    @given(
+        # Intentionally restricting the range of the rate to avoid overflow errors or NaNs vs Infs checks
+        rate=st.floats(min_value=0.01, max_value=1000, allow_infinity=False),
+        nper=st.floats(min_value=1, max_value=100, allow_infinity=False),
+        pmt=st.floats(min_value=-1000, max_value=-0.01, allow_infinity=False),
+        when=when_period_strategy,
+    )
+    @settings(verbosity=Verbosity.verbose)
+    def test_pv_interest_rate_sensitivity(
+        self,
+        rate: float,
+        nper: float,
+        pmt: float,
+        when: Literal[0, 1, "begin", "end"],
+    ) -> None:
+        """
+        Test that the present value is inversely proportional to the interest rate,
+        all other things being equal.
+        """
+        result = npf.pv(rate=rate, nper=nper, pmt=pmt, when=when)
+        expected = float(npf.pv(rate=rate + 0.1, nper=nper, pmt=pmt, when=when))
+        
+        # As interest rate increases, present value decreases
+        assert round(result, 4) >= round(expected, 4)
```

Comment:
I tried to use something like `result >= pytest.approx(expected)`, but `pytest.approx` doesn't work with floats and greater-than-or-equal-to operator.